core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -262,18 +252,20 @@ protected void inboundHeadersReceived(Metadata headers) {
       statsTraceCtx.clientInboundHeaders();
 
       boolean compressedStream = false;
-      String streamEncoding = headers.get(CONTENT_ENCODING_KEY);
-      if (fullStreamDecompression && streamEncoding != null) {
-        if (streamEncoding.equalsIgnoreCase("gzip")) {
-          setFullStreamDecompressor(new GzipInflatingBuffer());
-          compressedStream = true;
-        } else if (!streamEncoding.equalsIgnoreCase("identity")) {
-          deframeFailed(
-              Status.INTERNAL
-                  .withDescription(
-                      String.format("Can't find full stream decompressor for %s", streamEncoding))
-                  .asRuntimeException());
-          return;
+      if (fullStreamDecompression) {
+        String streamEncoding = headers.get(CONTENT_ENCODING_KEY);
+        if (streamEncoding != null) {
+          if (streamEncoding.equalsIgnoreCase("gzip")) {
+            enableFullStreamDecompressor();
+            compressedStream = true;
+          } else if (!streamEncoding.equalsIgnoreCase("identity")) {
+            deframeFailed(
```

Comment:
i was expecting a lookup in the registry. Am i right to assume that 'gzip' the only supported Codec?

Reply:
Yes. The integration needed here is quite strong, such that we can't use the registry.

Reply:
Somewhat sad. Thank you, anyway!